java8homework/src/main/java/Calculations.java

```diff
@@ -0,0 +1,155 @@
+        System.out.println(ColorInLife.ANSI_BLUE + "Total number of integers: " + randomInts.size() + "\n");
+    }
+
+    /**
+     * 6. Create a map of "n" (K,V) elements and print "how many" elements have value over 10 (the key is of your choice)
+     */
+    public static Map<String, Integer> topRanking(List<Person> listOfPerson, int score) {
+        Map<String, Integer> mappingKV = listOfPerson.stream()
+                .collect(Collectors.toMap(Person::getName, Person::getTennisScore));
+
+        System.out.println("Number of players who obtained over " + score + " points is: " + mappingKV.values().stream()
+                .filter(x -> x > score)
+                .count() + "\n");
+
+        return mappingKV;
+    }
+
+    /**
+     * 7. Sort the above Set<String> (used for max) in reverse order
+     */
+    public static Set<String> reversingOrder(Set<String> randomStrings) {
+        return new TreeSet<>(randomStrings).descendingSet();
+    }
+
+    /**
+     * 8. Sort the above List of custom objects (used for filter) in an order you consider
+     */
+    public static List<Person> ageSorting(List<Person> someList) {
+        return someList.stream()
+                .sorted(Comparator.comparing(Person::getAge))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * 9. Check if any of your instances "match" a condition based on an Integer field (of your choice).
+     * What does it return ? Print it out.
+     */
+    public static boolean scoreConditionValidation(List<Person> someList, int score) {
+        return someList.stream().anyMatch(Person -> Person.getTennisScore() > score);
+    }
+
+    /**
+     * 10. What does Optional represent ? Explain through an example on your custom object
+     * a) Wrap an existing instance
+     * b) Wrap a null
+     * c) Check value using ifPresent or isPresent
+     * <p>
+     * Optional is a container object used to contain not-null objects. Optional object is used to represent null with
+     * absent value. This class has various utility methods to facilitate code to handle values as ‘available’ or
+     * ‘not available’ instead of checking null values
+     */
+    public static void representingOptional(Person tennisMan, List<String> emptyList) {
+        Optional<Person> thePerson = Optional.of(tennisMan); // a)
+
+        Optional<String> wrapped = emptyList.stream() // b)
+                .min((String::compareTo));
+
+        // c)
+        thePerson.ifPresent(p -> System.out.println(tennisMan));
+
+        if (thePerson.isPresent()) {
+            System.out.println("The person exists\n");
+        } else {
+            System.out.println("No person exists\n");
+        }
+    }
+
+    /**
+     * 11. Fastest way to find the shortest String in a List
+     * (take the 5 random Strings generated above and find the shortest one).
+     */
+    public static String minString(Set<String> insertSet) {

```

Comment:
actually the fastest way is .reduce(), depends on the number of elements in the list and if it is sequential / parallel, you could do some tests to check this out